servers/visual/shader_language.cpp

```diff
@@ -3661,139 +3661,148 @@ Error ShaderLanguage::_parse_shader(const Map<StringName, FunctionInfo> &p_funct
 					//todo parse default value
 
 					tk = _get_token();
-					if (tk.type == TK_OP_ASSIGN) {
-
-						Node *expr = _parse_and_reduce_expression(NULL, Map<StringName, BuiltInInfo>());
-						if (!expr)
-							return ERR_PARSE_ERROR;
-						if (expr->type != Node::TYPE_CONSTANT) {
-							_set_error("Expected constant expression after '='");
-							return ERR_PARSE_ERROR;
-						}
-
-						ConstantNode *cn = static_cast<ConstantNode *>(expr);
-
-						uniform.default_value.resize(cn->values.size());
-
-						if (!convert_constant(cn, uniform.type, uniform.default_value.ptrw())) {
-							_set_error("Can't convert constant to " + get_datatype_name(uniform.type));
-							return ERR_PARSE_ERROR;
-						}
-						tk = _get_token();
-					}
+					while (true) {
```

Comment:
Why not just write `while (tk.type == TK_OP_ASSIGN)` ?

Reply:
No compelling reason, honestly.  I was just trying to follow existing conventions and `while (true)` was prevalent.  If that's the only issue, I can definitely address that.

Reply:
Well, it depends. But here it make the code a lot more longer and far less readable. So yeah, I would prefer replacing by `while (tk.type == TK_OP_ASSIGN)`. :)
Please amend your commit (no need to create a new one)

Reply:
Wait, it doesn't work.  The first time through the loop is the colon operator.  The second time through is the assignment.

Reply:
Arf sorry. My bad, I did not unfolded the code XD.

I would personally have put the `break` inside each `if`, or a least get rid of the case with the `continue` as it makes no sense at the end of the loop. But, anyway, that's ok like that, sorry for the inconvenience.